### PR TITLE
Updates countries_controller_test

### PR DESCRIPTION
**Pull request by adityathiru**
* Add tests for country search and country update
* Refactors countries_controller

### Checklist

- [x] I read the [contributing guide](/CONTRIBUTING.md) and I've followed it.
- [x] Commit message gives complete and useful information about the changes under it.
- [x] I ran the code locally and checked for common errors and uncaught exceptions.
- [x] My code is unit testable.

After you submit your pull request, DO NOT close it or click 'Update Branch'.

Occasionally, you may be asked to rebase your branch.

### CODE REVIEW

File: app/controllers/countries_controller.rb [hunk 1]
```diff
@@ -12,58 +14,65 @@ def index
...
-
+        @countries = Country.search(params[:search]).order("created_at DESC")
+      else
+        @countries = current_user.organization.countries.order('created_at DESC')
+      end
     end
   end
 
   def show
-    @country = Country.find(params[:id])
```
Comment (prachi1210) [inline app/controllers/countries_controller.rb]: @adityathiru Have you deprecated this function? 
In which case you should comment it out and add deprecated label? 
Same for edit.

File: app/controllers/countries_controller.rb [hunk 2]
```diff
@@ -1,5 +1,7 @@
 class CountriesController < ApplicationController
+  before_action :set_country, only: [:show, :edit, :update, :destroy]
```
Comment (adityathiru) [inline app/controllers/countries_controller.rb]: @prachi1210 I've created a before_action hook which initiates before the following functions are called. This was done just to make the code cleaner.

File: app/controllers/countries_controller.rb [hunk 3]
```diff
@@ -12,58 +14,65 @@ def index
...
-    @country = Country.find(params[:id])
     @country.destroy
 
+    flash[:notice] = "Country has been deleted."
     redirect_to countries_path
   end
 
   private
+  # Use callbacks to share common setup or constraints between actions.
+  def set_country
```
Comment (adityathiru) [inline app/controllers/countries_controller.rb]: @prachi1210 This is the code for the hook

**Comment by prachi1210** [comment]
PR seems good to me @adityathiru 
I'd just suggest adding comments for those functions so as to avoid confusion.

**Comment by prachi1210** [comment]
@adityathiru please make sure that you add relevant comments in your code when you send PR for documentation of phase 2 work

**Comment by adityathiru** [comment]
@prachi1210 Sure ma'am. I'll add necessary comments when documenting phase 2 work.

**Pull request merged**